packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -2,20 +2,158 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:meta/meta.dart';
+
 import 'basic_types.dart';
+import 'diagnostics.dart';
 import 'print.dart';
 
 /// Signature for [FlutterError.onError] handler.
 typedef FlutterExceptionHandler = void Function(FlutterErrorDetails details);
 
 /// Signature for [FlutterErrorDetails.informationCollector] callback
-/// and other callbacks that collect information into a string buffer.
-typedef InformationCollector = void Function(StringBuffer information);
+/// and other callbacks that collect information describing an error.
+typedef InformationCollector = void Function(List<DiagnosticsNode> information);
```

Comment:
since we're changing it anyway, would it make sense to make `InformationCollector` have this signature instead?:
```dart
typedef InformationCollector = Iterable<DiagnosticsNode> Function();
```
That way it could be used like this:
```dart
      informationCollector: () sync* {
        yield DiagnosticsProperty<PointerEvent>('Event', event, style: DiagnosticsTreeStyle.indentedSingleLine);
        yield DiagnosticsProperty<HitTestTarget>('Target', entry.target, style: DiagnosticsTreeStyle.indentedSingleLine);
      },
```
...which seems more idiomatic and terser without loss of generality or expressivity.

Reply:
sgtm! Refactoring now.

Reply:
refactored. the api looks much nicer with sync*